lib/data/repositories/collection_repository.dart

```diff
@@ -62,17 +73,39 @@ class CollectionRepositoryImpl implements CollectionRepository {
   }
 
   @override
-  Future<List<Stream<Collection>>> listenToAllCollections() async {
-    final rawCollections = await _db.getAll(store: _collectionStore);
-    final collectionIds = rawCollections.map((rawCollection) => rawCollection[CollectionKeys.id] as String).toList();
+  Future<List<CollectionMemos>> getCollectionMemosByIds(List<String> ids) async {
+    final collectionsFutures = ids.map((id) => _appBundle.loadJson('$_collectionsRoot/$id.json'));
+    final rawCollections = await Future.wait<dynamic>(collectionsFutures);
+
+    final castCollections = List<Map<String, dynamic>>.from(rawCollections);
+    return castCollections.map(_collectionsMemosSerializer.from).toList();
+  }
 
-    final listenFutures =
-        collectionIds.map((collectionId) => _db.listenTo(id: collectionId, store: _collectionStore)).toList();
-    final rawCollectionListeners = await Future.wait(listenFutures);
+  @override
+  Future<void> putCollectionsWithCollectionMemos(List<CollectionMemos> collections) async {
+    return _db.putAll(
+      ids: collections.map((collection) => collection.id).toList(),
+      objects: collections
+          .map(
+            (collection) => {
```

Comment:
Isn't this the work of the `CollectionSerializer`?

Reply:
Nope, it only updates part of the object. This is a very useful feature because one may not need to fetch an object before updating, so we can do partial updates and in a type-safe fashion, where we expose the fields by the keys (in this case the CollectionKeys)